### PR TITLE
callable redirect prop

### PR DESCRIPTION
**Pull request by fu2re**
redirect prop in forms or save button can be a function now.

### CODE REVIEW

**Comment by fu2re** [comment]
For example if you want set the recently added object as value of reference field in the another object.

File: docs/CreateEdit.md [hunk 1]
```diff
@@ -415,6 +415,28 @@ export const PostEdit = (props) => (
     </Edit>
 );
 ```
+In case when this is not enough you can also use a function. For example if you want set the recently added object as value of reference field in the another object:
+Unfortunately I do not know how to write test for that.
```
Comment (djhi) [inline docs/CreateEdit.md]: Please remove this line

File: src/util/resolveRedirectTo.js [hunk 2]
```diff
@@ -11,6 +11,9 @@ export default (redirectTo, basePath, id) => {
         case 'show':
             return `${linkToRecord(basePath, id)}/show`;
         default:
+            if (typeof redirectTo === 'function') {
```
Comment (djhi) [inline src/util/resolveRedirectTo.js]: Please add a unit test for this case in `resolveRedirectTo.spec.js`.
Pass a function as the `redirecTo` which returns its parameters and asserts they are corrects

**Comment by djhi** [comment]
Also please either squash your commits into one or give them a proper message

**Comment by fzaninotto** [comment]
Also, this is a new feature, so it should be opened against `next`, not `master`

**Comment by djhi** [comment]
ping @fu2re :)

**Comment by fu2re** [comment]
I ma trying right now make my changes right now against next. But I have some troubles with test dependencies

**Comment by djhi** [comment]
We'll need more details if you want some help :)

**Comment by fu2re** [comment]
FAIL  packages/react-admin/src/mui/input/ImageInput.spec.js
  ● Test suite failed to run

    Cannot find module './Mime' from 'index.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:194:17)
      at Object.<anonymous> (node_modules/mime/index.js:3:12)

when I try to make test

**Comment by fu2re** [comment]
I have created the new pull request aginst next and close the one.
https://github.com/marmelab/admin-on-rest/pull/1384

**Pull request closed without merging**